update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+        if cert is None:
+            sys.stdout.write("No certicate defined for %s " % hostname +
+                             "voms server.\n")
+            x509_cert = None
+        else:
+            try:
+                x509_cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
+
+            except crypto.Error as e:
+                sys.stderr.write("ERROR Certificate for %s " % hostname +
+                                 "could not be loaded: %s\n" % str(e))
+        old_x509_cert = None
+        if x509_cert is not None and os.path.isfile(server_cert_tpl):
+            if debug:
+                sys.stdout.write("DEBUG Retrieving VOMS server"
+                                 " %s existing certificate\n" % hostname)
+            template = open(server_cert_tpl, 'r')
+            old_cert = ""
+            in_cert_part = False
+            cert_start_reg = re.compile("^'cert' \?=")
+            cert_end_reg = re.compile("^EOF")
+            for line in template.readlines():
+                if in_cert_part:
+                    if re.search(cert_end_reg, line):
+                        in_cert_part = False
+                        break
+                    else:
+                        old_cert += line
+                elif re.search(cert_start_reg, line):
+                    in_cert_part = True
+            template.close()
+            if old_cert:
+                try:
+                    old_x509_cert = crypto.load_certificate(
+                        crypto.FILETYPE_PEM, old_cert
+                    )
+                except crypto.Error as e:
+                    sys.stderr.write("ERROR Existing certificate for " +
+                                     "%s could not be " % hostname +
+                                     "loaded: %s\n" % str(e))
+                    old_x509_cert = None
+                if old_x509_cert is not None:
+                    if old_x509_cert.has_expired():
+                        sys.stdout.write("Existing certificate" +
+                                         "(%s) is no " % server_cert_tpl +
+                                         "longer valid: ignoring it.\n")
+                        old_x509_cert = None
+                    elif (old_x509_cert.digest('SHA256')
+                          == x509_cert.digest('SHA256')):
+                        sys.stdout.write("Existing certificate " +
+                                         "(%s) is the " % server_cert_tpl +
+                                         "same than the retrieved " +
+                                         "certificate: ignoring it.\n")
+                        old_x509_cert = None
+        sys.stdout.write("Updating template for VOMS server " +
+                         "%s (%s)\n" % (hostname, server_cert_tpl))
+        template = open(server_cert_tpl, 'w')
+        template.write("structure template %s;\n\n" % server_cert_ns)
+        template.write("'cert' ?= <<EOF;\n")
```

Comment:
Could you write the certificates to seperate files and reference them with `file_contents()`?
1. It would be neater.
2. They could be checked by other tools more easily.

Reply:
To go further, the vo/certs dir could only contain certificates (named like ```voms.sagrid.ac.za.pem```) and the https://github.com/quattor/template-library-grid/blob/umd-4/vo/config.pan template modified to include this certificates (only a few line to modify). If you agree, I would open a _enhancement_ issue once this PR is merged.

Reply:
Sounds like a great idea to me!